File: strategy_java/src/test/java/com/kmmania/strategy_training/SprintTrainingTest.java
package com.kmmania.strategy_training;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.time.LocalDate;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

class SprintTrainingTest extends BaseTrainingTest {
    @Test
    @DisplayName("train() method should print SprintTraining content")
    void train_ShouldPrintSprintTrainingCotent() {
        // Given
        TrainingSession trainingSessionMock = mock(TrainingSession.class);
        when(trainingSessionMock.date()).thenReturn(LocalDate.now());
        when(trainingSessionMock.timeSlot()).thenReturn("Morning");
        when(trainingSessionMock.place()).thenReturn("Insep");

        SprintTraining sprintTraining = new SprintTraining(trainingSessionMock);

        // When
        sprintTraining.train();

        // Then
        assertThat(getStandardOutContent()).contains("*** Sprint training ***");
        assertThat(getStandardOutContent()).contains("Date: " + LocalDate.now());
        assertThat(getStandardOutContent()).contains("Time slot: Morning");
        assertThat(getStandardOutContent()).contains("Place: Insep");
    }
}

File: strategy_kotlin/src/test/kotlin/strategy_training/SprintTrainingTest.kt
package strategy_training

import com.kmmania.strategy_training.SprintTraining
import com.kmmania.strategy_training.TrainingSession
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.`when`
import java.time.LocalDate

class SprintTrainingTest : BaseTrainingTest() {
    @Test
    fun `train() method should print SprintTraining content`() {
        // Given
        val trainingSessionMock = mock(TrainingSession::class.java)
        `when`(trainingSessionMock.date).thenReturn(LocalDate.now())
        `when`(trainingSessionMock.timeSlot).thenReturn("Morning")
        `when`(trainingSessionMock.place).thenReturn("Insep")
        val sprintTraining = SprintTraining(trainingSessionMock)

        // When
        sprintTraining.train()

        // Then
        assertThat(getStandardOutContent()).contains("*** Sprint training ***")
        assertThat(getStandardOutContent()).contains("Date: ${LocalDate.now()}")
        assertThat(getStandardOutContent()).contains("Time slot: Morning")
        assertThat(getStandardOutContent()).contains("Place: Insep")
    }
}